pkg/apis/core/types.go

```diff
@@ -3776,6 +3779,27 @@ const (
 // ResourceList is a set of (resource name, quantity) pairs.
 type ResourceList map[ResourceName]resource.Quantity
 
+func (r *ResourceList) UnmarshalJSONPB(u *jsonpb.Unmarshaler, value []byte) error {
+	var m map[string]interface{}
+	err := json.Unmarshal(value, &m)
+	if err != nil {
+		return err
+	}
+
+	resourceList := make(ResourceList)
+	for resourceName, quantity := range m {
+		quantityMap := quantity.(map[string]interface{})
+		quantityIface, ok := quantityMap["string"]
+		if !ok {
+			return errors.New("missing string key in quantity")
+		}
+
+		resourceList[ResourceName(resourceName)] = resource.MustParse(quantityIface.(string))
```

Comment:
Since Unmarshal returns an error, I think we should use Parse and return the error if it failed to parse instead of panicing. We should also assert that quantityIface is of t ype string first
```suggestion
                 asString, casted := quantityIface.(string)
                 if !casted {
                      return fmt.Errorf("quantity string key doesn't have a string as a value. It has {} instead", reflect.TypeOf(quantityIface))
                 }
                 
                 
		resourceList[ResourceName(resourceName)], err = resource.Parse(asString)
		if err != nil {
		   // TODO: return a wrapped error if that's used somewhere else in this code base
		    return err 
		}
```